packages/core/src/api/blockManipulation/commands/getBlock/getBlock.ts

```diff
@@ -0,0 +1,127 @@
+import { Node } from "prosemirror-model";
+
+import { Block } from "../../../../blocks/defaultBlocks.js";
+import type { BlockNoteEditor } from "../../../../editor/BlockNoteEditor.js";
+import {
+  BlockIdentifier,
+  BlockSchema,
+  InlineContentSchema,
+  StyleSchema,
+} from "../../../../schema/index.js";
+import { nodeToBlock } from "../../../nodeConversions/nodeToBlock.js";
+import { getNodeById } from "../../../nodeUtil.js";
+
+// Get a block from a block identifier, or return undefined if the block cannot
+// be found in the editor. The ProseMirror node that gets converted into a block
+// is the one that has the same ID as the block identifier. However, if the
+// `getDifferentNodeForConversion` function is provided, the node returned from
+// it will be converted instead, e.g. a sibling or parent. If
+// `getDifferentNodeForConversion` returns undefined, this function will also
+// return undefined.
+function getBlockFromNode<
```

Comment:
I think it's cleaner to just remove this method and inline the code in `getPrevBlock` / `getParentBlock`, what do you think?

My thoughts are that the function does not really do a lot, and the other methods (`getNodeById` and `nodeToBlock` are pretty clear). While `getDifferentNodeForConversion` makes the overall code a bit shorter, using a callback to me adds unnecessary complexity that's not worth it